src/analysis/normed_space/dual.lean

```diff
@@ -115,3 +124,95 @@ def inclusion_in_double_dual_li : E →ₗᵢ[𝕜] (dual 𝕜 (dual 𝕜 E)) :=
 end bidual_isometry
 
 end normed_space
+
+section polar_sets
+
+open metric set normed_space
+
+/-- Given a subset `s` in a normed space `E` (over a field `𝕜`), the polar
+`polar 𝕜 s` is the subset of `dual 𝕜 E` consisting of those functionals which
+evaluate to something of norm at most one at all points `z ∈ s`. -/
+def polar (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
+  {E : Type*} [normed_group E] [normed_space 𝕜 E] (s : set E) : set (dual 𝕜 E) :=
+{x' : dual 𝕜 E | ∀ z ∈ s, ∥ x' z ∥ ≤ 1}
+
+namespace polar
+
+open metric set normed_space
+open_locale topological_space
+
+variables (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
+variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
+
+@[simp] lemma zero_mem (s : set E) :
+  (0 : dual 𝕜 E) ∈ polar 𝕜 s :=
+λ _ _, by simp only [zero_le_one, continuous_linear_map.zero_apply, norm_zero]
+
+lemma eq_Inter (s : set E) :
+  polar 𝕜 s = ⋂ z ∈ s, {x' : dual 𝕜 E | ∥ x' z ∥ ≤ 1} :=
+by { ext, simp only [polar, mem_bInter_iff, mem_set_of_eq], }
+
+lemma of_empty : polar 𝕜 (∅ : set E) = univ :=
```

Comment:
I don't think the namespace is a good idea here. There's no dot notation to be used. And the `.of_` is weird. What about `zero_mem_polar`, `polar_eq_Inter`, `polar_empty`, `smul_mem_polar`, `polar_closed_ball_one`, ... ?

Reply:
All of these sound good, except that `ball_one` instead of `unit_ball` sounded so strange that I had to generalize to an arbitrary positive radius :D. (Ok, that is what I obviously should have been done in the first place...) I changed others as suggested.